engine/src/main/java/com/datatorrent/stram/StreamingContainerAgent.java

```diff
@@ -271,10 +282,9 @@ public void addOperatorRequest(StramToNodeRequest r) {
 
         // On the input side there is a unlikely scenario of partitions even for inline stream that is being
         // handled. Always specifying a stream codec configuration in case that scenario happens.
-        InputPortMeta idInputPortMeta = getIdentifyingInputPortMeta(in);
-        StreamCodec<?> streamCodecInfo = getStreamCodec(idInputPortMeta);
-        Integer id = physicalPlan.getStreamCodecIdentifier(streamCodecInfo);
-        inputInfo.streamCodecs.put(id, streamCodecInfo);
+        final StreamCodec<?> streamCodec = getIdentifyingInputPortMeta(in).getStreamCodec();
```

Comment:
I don't think it is necessary to refactor the when stream codec is being computed as the main intent of this change is to fix the attribute map sharing problem. Could you limit the change just to cloning of the attributes. You could do the refactoring of stream codec handling as a separate item (probably in master only). Also, it should be possible for a non operator developer to override the stream codec specified by the operator developer in the input port with an attribute, the current refactoring does not allow that.

Also could you include the JIRA number.